syntax = "proto3";

package kvstore;

service KVStore {
  rpc Get(GetReq) returns (GetResp) {}
  rpc Put(PutReq) returns (PutResp) {}
  rpc Delete(DeleteReq) returns (DeleteResp) {}
}

enum ErrorCode {
  OK = 0;
  CLIENT_ERROR = 1;
  SERVER_ERROR = 2;
}

message Status {
  ErrorCode error_code = 1;
  bytes error_msg = 2;
}

message GetReq {
  bytes key = 1;
}

message GetResp {
  bytes value = 1;
  Status status = 2;
}

message PutReq {
  bytes key = 1;
  bytes value = 2;
}

message PutResp {
  Status status = 2;
}

message DeleteReq {
  bytes key = 1;
}

message DeleteResp {
  Status status = 2;
}